compiler/noirc_evaluator/src/ssa/opt/remove_enable_side_effects.rs

```diff
@@ -80,8 +88,8 @@ impl Function {
                 if condition_is_one {
                     last_side_effects_enabled_instruction = None;
                 } else {
-                    last_side_effects_enabled_instruction = Some(instruction_id);
                     context.remove_current_instruction();
+                    last_side_effects_enabled_instruction = Some(context.instruction_id);
```

Comment:
I'm seeing various regressions and this is the only place where it looks like that could have been caused.

Reply:
Or maybe it was this, not sure. The benchmarks looks good now (they are gone).